src/files/coverageparser.ts

```diff
@@ -17,147 +17,293 @@ export class CoverageParser {
 
             try {
                 parseContentJacoco(xmlFile, async (err, data) => {
                     checkError(err);
-                    const sections = await this.convertSectionsToMap(data);
-                    return resolve(sections);
+                    await this.addSections(coverages, data);
+                    return resolve();
                 });
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
             } catch (error: any) {
                 checkError(error);
             }
         });
     }
 
-    private async xmlExtractClover(filename: string, xmlFile: string) {
+    private async xmlExtractClover(
+        coverages: Map<string, Section>,
+        coverageFilename: string,
+        xmlFile: string
+    ) {
         try {
             const data = await parseContentClover(xmlFile);
-            const sections = await this.convertSectionsToMap(data);
-            return sections;
-        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            await this.addSections(coverages, data);
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
         } catch (error: any) {
-            error.message = `filename: ${filename} ${error.message}`;
+            error.message = `filename: ${coverageFilename} ${error.message}`;
             this.handleError("clover-parse", error);
             return new Map<string, Section>();
         }
     }
 
-    private lcovExtract(filename: string, lcovFile: string) {
-        return new Promise<Map<string, Section>>((resolve) => {
+    private lcovExtract(
+        coverages: Map<string, Section>,
+        coverageFilename: string,
+        lcovFile: string
+    ) {
+        return new Promise<void>((resolve) => {
             const checkError = (err: Error) => {
                 if (err) {
-                    err.message = `filename: ${filename} ${err.message}`;
+                    err.message = `filename: ${coverageFilename} ${err.message}`;
                     this.handleError("lcov-parse", err);
-                    return resolve(new Map<string, Section>());
+                    return resolve();
                 }
             };
 
             try {
                 source(lcovFile, async (err, data) => {
                     checkError(err);
-                    const sections = await this.convertSectionsToMap(data);
-                    return resolve(sections);
+                    await this.addSections(coverages, data);
+                    return resolve();
                 });
-            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+                // eslint-disable-next-line @typescript-eslint/no-explicit-any
             } catch (error: any) {
                 checkError(error);
             }
         });
     }
 
+    private mergeSections(existingSection: Section, section: Section): Section {
```

Comment:
[F] If you look at https://github.com/ryanluker/vscode-coverage-gutters/pull/483/commits/66df5d7bc3a00d76dd0406072a339c23e194cc56 you can see how I initially worked this out by updating the `node` example project.